php/pantheon/checks/themes.php

```diff
@@ -54,42 +57,74 @@ public function run() {
 
 			$data = wp_get_theme($slug);
 			$version = $data->version;
-			$vulnerable = $this->is_vulnerable($slug, $version);
+			if ( $should_check_vulnerabilities ) {
+				$vulnerable = $this->is_vulnerable($slug, $version);
+			}
 
 			$needs_update = 0;
 			$available = '-';
 			if (isset($update[$theme_path])) {
 				$needs_update = 1;
 				$available = $update[$slug]->update["new_version"];
 			}
-			if ( false === $vulnerable ) {
-				$vulnerable = "None";
-			} else {
+			if ( $should_check_vulnerabilities && $vulnerable ) {
 				$vulnerable = sprintf('<a href="https://wpscan.com/themes/%s" target="_blank" >more info</a>', $slug );
+			} elseif ( $should_check_vulnerabilities ) {
+				$vulnerable = "None";
 			}
 
 			$report[$slug] = array(
 				'slug' => $slug,
 				'installed' => (string) $version,
 				'available' => (string) $available,
 				'needs_update' => (string) $needs_update,
-				'vulnerable'  => $vulnerable,
 			);
+
+			if ( $should_check_vulnerabilities ) {
+				$report[ $slug ]['vulnerable'] = $vulnerable;
+			}
 		}
 		$this->alerts = $report;
 	}
 
+	/**
+	 * Get a WordPress vulnerability API token if one is defined and we're in the right environment.
+	 * Copied from wp_launch_check/php/pantheon/checks/plugins.php
+	 * Uses the WPSCAN_API_TOKEN constant if defined.
+	 *
+	 * @return string|false
+	 * @todo Replace this with a Patchstack API token.
+	 */
+	protected function getWpVulnApiToken() {
```

Comment:
why is this method not somewhere it can be used by both checks? abstract/refactor this out.

Reply:
Overall would love to see a class sit between `Plugins`|`Themes` and `Checkimplementation` that covers the common code.